recipes/msix/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+    }
+
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/**"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        if self.settings.os == "Android":
+            self._cmake.definitions["AOSP"] = True
+        if self.settings.os == "Linux":
+            self._cmake.definitions["LINUX"] = True
+        if self.settings.os == "Macos":
+            self._cmake.definitions["MACOS"] = True
+        self._cmake.definitions["CRYPTO_LIB"] = self.options.crypto_lib
+        self._cmake.definitions["MSIX_PACK"] = self.options.pack
+        self._cmake.definitions["MSIX_SAMPLES"] = False
+        self._cmake.definitions["MSIX_TESTS"] = False
+        self._cmake.definitions["SKIP_BUNDLES"] = self.options.skip_bundles
+        self._cmake.definitions["USE_EXTERNAL_ZLIB"] = self.options.use_external_zlib
+        self._cmake.definitions["USE_SHARED_ZLIB"] = self.options["zlib"].shared
+        self._cmake.definitions["USE_VALIDATION_PARSER"] = self.options.use_validation_parser
+        self._cmake.definitions["XML_PARSER"] = self.options.xml_parser
+        self._cmake.configure()
+        return self._cmake
+
+    def validate(self):
+        if self.settings.os != "Android" and self.options.xml_parser == "javaxml":
+            raise ConanInvalidConfiguration("javaxml is supported only for Android")
+        if self.settings.os == "Linux" and self.settings.compiler != "clang":
+            raise ConanInvalidConfiguration("Only clang is supported on Linux")
+        if self.settings.os != "Macos" and self.options.xml_parser == "applexml":
+            raise ConanInvalidConfiguration("applexml is supported only for MacOS")
+        if self.settings.os != "Windows" and self.options.crypto_lib == "crypt32":
+            raise ConanInvalidConfiguration("crypt32 is supported only for Windows")
+        if self.settings.os != "Windows" and self.options.xml_parser == "msxml6":
+            raise ConanInvalidConfiguration("msxml6 is supported only for Windows")
+        if self.options.pack:
+            if self.settings.os == "Macos":
+                if not self.options.use_external_zlib:
+                    raise ConanInvalidConfiguration("Using libCompression APIs and packaging features is not supported")
+                if self.options.xml_parser != "xerces":
+                    raise ConanInvalidConfiguration("Xerces is the only supported parser for MacOS pack")
+            if not self.options.use_validation_parser:
+                raise ConanInvalidConfiguration("Packaging requires validation parser")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Linux" and self.options.shared:
+            self.options.fPIC = True
```

Comment:
```suggestion
        if self.options.shared:
            del self.options.fPIC
```
If the reason you're forcing `fPIC` to true when building shared is because the linker complained some objects not being compiled with `-fPIC`, then you should do this instead in cmake:
```
self._cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
```

It's not ideal behavior by conan.
@SpaceIm What do you think?

Reply:
I agree, inject `CMAKE_POSITION_INDEPENDENT_CODE` and delete PIC if shared. shared is always PIC

Reply:
This behavior is the garbage left from the time when we used the vendored `xerces-c` library. Not needed anymore after switching to the CCI package, [done](https://github.com/conan-io/conan-center-index/pull/6475/commits/2290f0ac84616d6c48b5290ac7036e0e1b120803)